api/annotations.py

```diff
@@ -149,9 +149,6 @@ def detail(cls, annotation, with_context=True):
 class AnnotationParser:
     @classmethod
     def parse(cls, _db, data, patron):
-        if patron.synchronize_annotations != True:
-            return PATRON_NOT_OPTED_IN_TO_ANNOTATION_SYNC
```

Comment:
We should be able to remove the `PATRON_NOT_OPTED_IN_TO_ANNOTATION_SYNC` ProblemDetail as well, since nothing else will be using it.